[20.09] always mount a job tmp rw

bernt-matthias opened the pull request:
Fixes: 

- In containerized tool runs `$_GALAXY_JOB_TMP_DIR` is created, the variable is "exported", but the directory **is not mounted writable**. 
- add `export _GALAXY_JOB_TMP_DIR` to make it usable in non-containerized tool runs
- `export GALAXY_SLOTS` as part of the cluster slot statement

bernt-matthias committed:
export _GALAXY_JOB_HOME_DIR and _GALAXY_JOB_TMP_DIR

these two variables are available in containers because of

`docker run ...-e "_GALAXY_JOB_HOME_DIR=$_GALAXY_JOB_HOME_DIR" -e "_GALAXY_JOB_TMP_DIR=$_GALAXY_JOB_TMP_DIR"`

but since they are not exported the same does not hold for non containerised
jobs.

Also moved `export GALAXY_SLOTS` to CLUSTER_SLOTS_STATEMENT.sh. Unsure
if this export should be changed to something more complex as in
MEMORY_STATEMENT.sh.

bernt-matthias committed:
add docs to xsd

bernt-matthias committed:
always mount a job tmp rw

bernt-matthias committed:
test writing to job home and tmp

bernt-matthias committed:
Remove doc of _GALAXY_JOB_HOME_DIR

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

bernt-matthias committed:
Remove superficial export of _GALAXY_JOB_HOME_DIR

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

mvdbeek commented on test/functional/tools/job_environment_default.xml:
```diff
@@ -3,14 +3,17 @@
+      echo "\$TMP"  > '$tmp' &&
+      echo "\$SOME_ENV_VAR" > '$some_env_var' &&
+      ## test if job home and tmp are mounted rw
+      touch '\$_GALAXY_JOB_TMP_DIR/test' &&
```
```suggestion
      touch "\$_GALAXY_JOB_TMP_DIR/test" &&
```

mvdbeek commented on test/functional/tools/job_environment_default.xml:
```diff
@@ -3,14 +3,17 @@
+      echo "\$SOME_ENV_VAR" > '$some_env_var' &&
+      ## test if job home and tmp are mounted rw
+      touch '\$_GALAXY_JOB_TMP_DIR/test' &&
+      touch '\$_GALAXY_JOB_HOME_DIR/test'
```
```suggestion
      touch "\$_GALAXY_JOB_HOME_DIR/test"
```

mvdbeek reviewed (commented):
You need to double quote environment variables

bernt-matthias committed:
fix quoting

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

bernt-matthias committed:
simplify debugging

bernt-matthias committed:
Revert "Remove superficial export of _GALAXY_JOB_HOME_DIR"

This reverts commit 365fb066b0d8c86f4d45a3e626969756506c315a.

bernt-matthias committed:
fix unit test

the export of GALAXY_SLOTS is now part of the slots_statement

bernt-matthias committed:
fix chained dynamic destinations integration test

- create a tmp dir as absolute path
- actually create a dir in the dyndest test

bernt-matthias committed:
create a test file in $HOME

bernt-matthias committed:
Fix another integration test

TmpDirAsShellCommandJobEnvironmentIntegrationTestCase.test_default_environment_1801

- actually create a dir
- create the dir in the job working dir
